lib/src/main/kotlin/TreeMap.kt

```diff
@@ -0,0 +1,76 @@
+package TreeMap
+
+abstract class Node<K: Comparable<K>, V, N>(key: K, value: V, parent: N?){
+    internal var key: K = key
+    internal var value: V = value
+    internal var parent: N? = parent
+
+    internal var leftChild: N? = null
+
+    internal var rightChild: N? = null
+}
+
+
+abstract class TreeMap<K: Comparable<K>, V, N: Node<K, V, N>>: Iterable<Pair<K, V>> {
+    public var size: Int = 0
+    protected abstract var root: N?
+
+    public fun contains(key: K): Boolean {
+        var curNode = this.root
+        if (curNode == null) {
+            return false
+        }
+
+        while (curNode != null) {
+            if (curNode.key == key) {
+                return true
+            }
+            else if (curNode.key > key) {
+                curNode = curNode.leftChild
+            }
+            else if (curNode.key < key) {
+                curNode = curNode.rightChild
+            }
+        }
+
+        return false
+    }
```

Comment:
По сути contains дублирует код find, 
```suggestion
    public fun contains(key: K): Boolean { 
           return find(key) != null
    }

```

Reply:
предполагалось (правда, было сделано не так), что find будет или возвращать значение по ключу, или (когда ключа нет в дереве) выбрасывать NoSuchElementException, а contains будет говорить о том, есть ли ключ в дереве. Будет учтено